assets/scripts/build_gallery.py

```diff
@@ -320,10 +327,46 @@ def get_expectation_instances(expectations_info):
                 f"\n\n----------------\n{expectation_name} ({expectations_info[expectation_name]['package']})\n"
             )
             expectation_tracebacks.write(traceback.format_exc())
-            continue
+        except pydantic.ValidationError:
+            expectation_tracebacks.write(traceback.format_exc())
+        except IndexError:
+            expectation_tracebacks.write(
+                f"Expectation {expectation_name} has invalid test case."
+            )
+            expectation_tracebacks.write(traceback.format_exc())
+        except Exception as exc:
+            expectation_tracebacks.write(f"Unexpected error occurred: {exc}")
+            expectation_tracebacks.write(traceback.format_exc())
     return expectation_instances
 
 
+def create_expectation_instance(expectation_class: type[Expectation]) -> Expectation:
+    """Accepts an Expectation class and attempts to instantiate it with valid parameters
+    obtained from the Expectation's test cases.
+    """
+    expectation_params: dict
+    if expectation_class.examples:
+        # take the first test case available:
+        expectation_params = expectation_class.examples[0]["tests"][0]["in"]
+
+    else:
+        _TEST_DEFS_DIR: Final = pathlib.Path(
+            __file__, "..", "..", "..", "tests", "test_definitions"
+        ).resolve()
+        found = next(
+            _TEST_DEFS_DIR.rglob(f"**/{expectation_class.expectation_type}.json"), None
+        )
+        if not found:
+            raise Exception(
+                f"No JSON test case found for Expectation {expectation_class.expectation_type}, cannot build gallery example."
+            )
+        with open(found) as fp:
+            test_cases = json.load(fp)
+            # take the first test case available:
+            expectation_params = test_cases["datasets"][0]["tests"][0]["in"]
```

Comment:
Same note as above - perhaps we check if "catch_exceptions" or another similar key is around